src/Microsoft.AspNetCore.SignalR/Proxies.cs

```diff
@@ -54,12 +55,28 @@ public Task InvokeAsync(string method, params object[] args)
         }
     }
 
+    public class AllClientsExceptProxy<THub> : IClientProxy
+    {
+        private readonly HubLifetimeManager<THub> _lifetimeManager;
+        private readonly List<string> _excludedIds;
+
+        public AllClientsExceptProxy(HubLifetimeManager<THub> lifetimeManager, List<string> excludedIds)
```

Comment:
The point is moot until we have benchmarks to properly understand the performance characteristics in practical scenarios. Given that the most common scenario we've seen for this involves a very small number of entries in the list, we may just spin our wheels here for a while with almost no gain :).

The original comment is just about using an interface over the concrete `List<T>` type currently used. `IReadOnlyCollection<T>`/`IReadOnlyList<T>` are the "correct" expressions of what this public API requires, which is generally our first stop when building an API. Performance decisions to compromise that have to be data-driven and based on actual usage.